### PR TITLE
Drop dependency on deprecated `gulp-util`

### PR DESCRIPTION
**Pull request by demurgos**
Closes dlmanning/gulp-sass#646

### CODE REVIEW

**Comment by demurgos** [comment]
This PR also update the versions tested with CI to match the ones used by Sass itself.
The main difference is that `0.10.0` is no longer tested because Node stopped maintaining it, and most dependencies started to require Node 4+.

File: .travis.yml [hunk 1]
```diff
@@ -1,6 +1,7 @@
 language: node_js
 node_js:
-  - "0.10"
```
Comment (xzyfer) [inline .travis.yml]: Node sass still officiallly supports 0.10, 0.12, and iojs. It's not tested in Travis but is tested locally before releases.

**Comment by xzyfer** [comment]
Thanks. Dropped support for old node versions will require a major version bump. I wasn't planning on bumping the major until node-sass@5 landed. Is it possible to make this work with Node 0.10?

**Comment by demurgos** [comment]
Hi,
Thanks for the info: I did not know that Sass still wants to support these versions. I'll try to find some older versions of the dependencies that support Node `0.10.0`. I [opened an issue](https://github.com/dlmanning/gulp-sass/issues/648) to discuss this change.

Regarding the lockfiles, you are better off by keeping them, especially if you plan to support unmaintained Node versions. The lock files are intended to be committed and have no effect on consumers: they are used for library developers and CI to make sure that you develop with the same working dependencies.

**Comment by demurgos** [comment]
Hi, I am having trouble to make `node-sass` work with iojs (Node 3). I pinned `node-sass` to exactly `4.2.0` but its install step has a dependency on request which itself depends on hoek which requires Node 4.

**Comment by xzyfer** [comment]
Correct. The latest release 4.7.2 pinned to a version on request that still
supported old node. Unfortunately that release has issues with iojs v3 so
I'm ok with not testing iojs in ci.

**Comment by demurgos** [comment]
@xzyfer
I tried my best to support it but at the moment, the best would be to merge #649 first, it's just going to cause more issues.
You are effectively asking me to drop support for iojs, so it's better to do it properly in its own PR and then come back to this issue.

Even if this Node issue is not related to `gulp-util`, it should be fixed first. Please also note that dropping `gulp-util` is important because it is causing build issues with the latest release of Gulp 4. You are not the first person I am contacting to patch this and multiple other packages have similar issues were old versions of Node are wrecking havoc the builds because they weren't touched in a year but most dependencies have moved on. We reached a point were not keeping dependencies up-to-date is causing builds to break so I'd really like to have it sorted as fast as possible. Thanks for your support :)

**Comment by demurgos** [comment]
I was wrong about lock files, I'll remove them.
Still, please consider migrating to newer Node versions and dropping gulp-util.

**Comment by demurgos** [comment]
@xzyfer @dlmanning 
Any news about this PR? `gulp-sass` is one of the most popular gulp plugins so it is important to keep its dependencies up-to-date.

**Comment by demurgos** [comment]
@xzyfer @dlmanning 
Are you looking for maintainers? We're currently discussing in the Gulp team to eventually create an org for popular plugins so we'd be able to help with this kind of maintenance.

**Comment by xzyfer** [comment]
We're happy to do it. We're just juggling release timelines for gulp-sass,
node-sass, and LibSass so that we're not leaving users behind with new
features landing in node-sass.

Dropping Node < 4 support is massive breaking change for us that requires
some coordination.

**Comment by betorobson** [comment]
hi, any update about this PR?

**Comment by xzyfer** [comment]
No. Any updates would appear on this issue.

**Comment by dmitry-vakhnenko** [comment]
Sorry, but what a shitty version choosing. node 4.* is fucking LEGACY. Just  bump to next major version, like it does ALL OTHERS  MOST POPULAR PACKAGES.

Sorry for ignore basic contribution rules but this is awesome.

**Comment by dmitry-vakhnenko** [comment]
@xzyfer will it be fixed after 5 years (now) or we just wait node 4 EOL?

**Comment by johnturingan** [comment]
@xzyfer @dlmanning no luck on merging this one? :(

**Comment by xzyfer** [comment]
Looks like there are a couple separate breaking changes that'll land soonish.

**Comment by rejas** [comment]
any news if this will get merged or if a new PR to fix the merge conflicts is necessary, @demurgos @xzyfer @dlmanning ?

**Comment by xzyfer** [comment]
There is linked pr that includes these changes and more and a checklist for
when it merged.

**Comment by xzyfer** [comment]
Thanks heaps, this has been included in #667

**Pull request closed without merging**